mmv1/main.go

```diff
@@ -140,46 +140,27 @@ func main() {
 	log.Printf("Building %s version", *version)
 	log.Printf("Building %s provider", providerName)
 
-	// Building compute takes a long time and can't be parallelized within the product
-	// so lets build it first
-	sort.Slice(allProductFiles, func(i int, j int) bool {
-		if allProductFiles[i] == "products/compute" {
-			return true
-		}
-		return false
-	})
-
-	var providerToGenerate provider.Provider
-
-	productFileChannel := make(chan string, len(allProductFiles))
 	productsForVersionChannel := make(chan *api.Product, len(allProductFiles))
-	for _, pf := range allProductFiles {
-		productFileChannel <- pf
-	}
-
-	for i := 0; i < len(allProductFiles); i++ {
+	for _, productFile := range allProductFiles {
 		wg.Add(1)
-		go GenerateProduct(productFileChannel, providerToGenerate, productsForVersionChannel, startTime, productsToGenerate, *resourceToGenerate, *overrideDirectory, generateCode, generateDocs)
+		go GenerateProduct(productFile, productsForVersionChannel, startTime, productsToGenerate, *resourceToGenerate, *overrideDirectory, generateCode, generateDocs)
 	}
 	wg.Wait()
 
-	close(productFileChannel)
 	close(productsForVersionChannel)
 
 	var productsForVersion []*api.Product
-
 	for p := range productsForVersionChannel {
 		productsForVersion = append(productsForVersion, p)
 	}
-
 	slices.SortFunc(productsForVersion, func(p1, p2 *api.Product) int {
 		return strings.Compare(strings.ToLower(p1.Name), strings.ToLower(p2.Name))
 	})
 
 	// In order to only copy/compile files once per provider this must be called outside
 	// of the products loop. This will get called with the provider from the final iteration
 	// of the loop
```

Comment:
```suggestion
	// of the products loop. Create an MMv1 provider with an arbitrary product (the first loaded).
```

Comment appears to have been stale already before your change, let's update that.